engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2246,6 +2306,41 @@ public LogicalOperatorInfo getLogicalOperatorInfo(String operatorName)
     return fillLogicalOperatorInfo(operatorMeta);
   }
 
+  public LogicalModuleInfo getLogicalModuleInfo(String moduleName, boolean flatten, LogicalPlan dag)
+  {
+    ModuleMeta moduleMeta = getModuleMeta(moduleName);
+    if (moduleMeta == null) {
+      return null;
+    }
+    LogicalModuleInfo logicalModuleInfo = fillLogicalModuleInfo(moduleMeta, flatten);
+    if (flatten) {
+      for (ModuleMeta meta : moduleMeta.getDag().getAllModules()) {
+        if (meta.getParentModuleName() != null && meta.getParentModuleName().equals(moduleName)) {
+          logicalModuleInfo.modules.add(getLogicalModuleInfo(fillLogicalModuleInfo(meta, flatten).name, flatten,
+              moduleMeta.getDag()));
+        }
+      }
+    }
+    return logicalModuleInfo;
+  }
+
+  public ModuleMeta getModuleMeta(String moduleName)
+  {
+    return getModuleMeta(moduleName, getLogicalPlan());
+  }
+
+  private ModuleMeta getModuleMeta(String moduleName, LogicalPlan dag)
```

Comment:
Private method is a utility method that is supposed to find ModuleMeta for a given module recursively. 
Fixed the logic.